Add some detail to docstrings for special functions

moble opened the pull request:
The `bi_zeros` and `pbvv_seq` docstrings were incorrectly copied from `ai_zeros` and
`pbdv_seq`, respectively.  This corrects those docstrings and adds some more
detail.

The `specfun` functions all come from "Computation of Special Functions" by
Zhang and Jin, so I've added a reference to that work to each function that
uses `specfun`.

This closes #4956 and closes #4957.

ev-br commented:
Thanks! Can you also run `cd special; ./generate_unfuncs.py` and commit the result (an appropriate commit message would be "GEN: regenerate _ufuncs.pyx"). Here's an explanation from Pauli of what this is: https://github.com/scipy/scipy/pull/2905#issuecomment-24835960

As a nitpick, these all functions cannot all come from Chapter 13 of the Zhang and Jun book, can they? If you have a book, it could be a bonus to add actual chapter numbers, but otherwise it's already an improvement.

moble commented:
Sorry, I thought I removed those chapters when pasting, but I guess I forgot.  That's not a nitpick at all.  :)    I've corrected some, and removed others.  Also added some `Parameters` sections to a few functions I wasn't clear on, and ran `generate_unfuncs.py`.

ev-br commented on scipy/special/basic.py:
```diff
@@ -1065,6 +1628,13 @@ def kelvin_zeros(nt):
     length 8 tuple of arrays of length nt.
     The tuple containse the arrays of zeros of
```
typo: containSE

moble commented:
I've gone through, and shortened all the first lines.  Some of those "first line"s were huge; I preserved the information by just moving it lower in the docstring, while still trying to make the summary sufficiently descriptive.  I haven't checked literally everything, but it compiles and everything I've looked at renders properly.

ev-br commented:
LGTM. I intend to merge it in a couple of days unless there are further comments. 

In an ideal world, there's a verified reference to A&S and/or DLMF for each special function, but that's clearly out of scope for this PR :-).

moble commented:
> In an ideal world, there's a verified reference to A&S and/or DLMF for each special function,

It'd be nice to have, since they're online and free.  But it's important to note that these `specfun` functions really come from Zhang and Jin, and they don't use the same notation as DLMF.  For example, the parabolic cylinder functions use conventions that I can't find elsewhere -- even DLMF's [listing of alternative notations](http://dlmf.nist.gov/12.1) isn't consistent with the usage of Zhang and Jin (and hence `scipy.special`).  So if anyone does come along and add those references later, it'd be good to point out that the notation may not be the same, or even figure out when the notation is and is not the same.

> but that's clearly out of scope for this PR :-).

Agreed.  :)

ev-br commented:
I do not disagree with you, adding the reference to Zhang&Jin is a clear improvement. All I'm saying is that the facts that the notation is different and that the book is not very easy to find IMO makes going over the usage and adding conversion formulas only more valuable.

ev-br commented:
OK, no objections for six days, I'll take it as a silent yes. Merging. Thank you @moble, keep them coming!
